dissect/util/stream.py

```diff
@@ -86,6 +91,8 @@ def read(self, n: int = -1) -> bytes:
         size = self.size
         align = self.align
 
+        self._seek_locked = True
```

Comment:
If any error occurs while reading, seeking will be locked forever. Perhaps it's better to work with a context manager or try/except? Or use [`threading.Lock`](https://docs.python.org/3/library/threading.html#threading.Lock) as a context manager.